pkg/endpointslice/map.go

```diff
@@ -0,0 +1,70 @@
+package endpointslice
+
+import (
+	"sync"
+
+	"github.com/submariner-io/lighthouse/pkg/constants"
+	discovery "k8s.io/api/discovery/v1beta1"
+)
+
+type endpointInfo struct {
+	key     string
+	hostIPs map[string][]string
+}
+
+type Map struct {
+	epMap map[string]*endpointInfo
+	sync.RWMutex
+}
+
+func (m *Map) GetIPs(hostname, cluster, namespace, name string) ([]string, bool) {
+	key := keyFunc(namespace, name, cluster)
+
+	result, ok := m.epMap[key]
+	if !ok {
+		return nil, false
+	}
+
+	ips, ok := result.hostIPs[hostname]
+
+	return ips, ok
```

Comment:
```suggestion
	return result.hostIPs[hostname]
```

Reply:
I had originally tried this way, it gives error `Not enough arguments to return`

Reply:
hmm that's weird - the two should be equivalent. Not a big deal.